### DISCUSSION TITLE
Creating a compound shape composed of multiple convex hulls

### DISCUSSION THREAD

**Discussion by SSNTails** [jrouwe/JoltPhysics#1017] [Q&A] [answered]
I have been able to successfully create a variety of shapes - spheres, boxes, capsules, convex hulls (singular). Now I want to create a Compound shape composed of multiple convex hulls.. the purpose of this would be to create a volume of water, for example, and be able to tell on each Update() whether the player is inside the water or not.

I feel like I might just be making one small mistake here that causes the shape to not be properly created or added to the physics system. I tried to find a sample in the repository, but the only compound shape examples I could find were for doing shape or ray cast tests.

`
JPH::StaticCompoundShapeSettings css;
css.mUserData = (JPH::uint64)object;

// Create compond convex hull using VHACD
VHACD::IVHACD *iface = VHACD::CreateVHACD();
Model::VHACDInfo *info = object->model->GetDataForVHACD();

VHACD::IVHACD::Parameters params;
params.m_findBestPlane = true;

iface->Compute(info->vertices, info->numVertices, info->indices, info->numTriangles, params);

uint32_t j;
for (j = 0; j < iface->GetNConvexHulls(); j++)
{
	VHACD::IVHACD::ConvexHull ch;
	if (!iface->GetConvexHull(j, ch))
		continue;

	JPH::Array<JPH::Vec3> vertexList;
	for (VHACD::Vertex v : ch.m_points)
		vertexList.push_back(JPH::Vec3((float)v.mX, (float)v.mY, (float)v.mZ));

	JPH::RefConst<JPH::Shape> convex = JPH::ConvexHullShapeSettings(vertexList, 0.0f).Create().Get();
	css.AddShape(mat.GetTranslation(), mat.GetRotation().GetQuaternion(), convex);
}

delete info;

JPH::BodyCreationSettings bcs(css.Create().Get(), mat.GetTranslation(), mat.GetRotation().GetQuaternion(), JPH::EMotionType::Kinematic, Layers::POLYOBJS);
bcs.mUserData = (JPH::uint64)object;
bcs.mRestitution = 0;
bcs.mAngularDamping = 0;
bcs.mFriction = 0;
bcs.mMaxAngularVelocity = 0;

bcs.mOverrideMassProperties = JPH::EOverrideMassProperties::MassAndInertiaProvided;
bcs.mMassPropertiesOverride.SetMassAndInertiaOfSolidBox(JPH::Vec3(1, 1, 1), 100);

object->collisionObjectID = bi.CreateAndAddBody(bcs, JPH::EActivation::Activate);`

**Comment by jrouwe** [comment, marked as answer]
It looks like you're double transforming by using mat.GetTranslation/Rotation on both the body and each of the subshapes

**Reply by SSNTails** [reply]
Oh dear, what a simple, stupid mistake! Thank you.
